hasher-matcher-actioner/hmalib/lambdas/hashing.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import functools
+from hmalib.common.content_sources import URLContentSource
+import boto3
+import os
+import json
+
+from mypy_boto3_dynamodb import DynamoDBServiceResource
+from mypy_boto3_sqs import SQSClient
+
+from threatexchange.content_type.meta import get_content_type_for_name
+from threatexchange.content_type.photo import PhotoContent
+from threatexchange.content_type.video import VideoContent
+from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.signal_type.pdq import PdqSignal
+
+from hmalib.common.logging import get_logger
+from hmalib import metrics
+from hmalib.common.message_models import URLSubmissionMessage
+from hmalib.hashing import Hasher
+
+logger = get_logger(__name__)
+sqs_client = boto3.client("sqs")
+
+
+@functools.lru_cache(maxsize=None)
+def get_dynamodb() -> DynamoDBServiceResource:
+    return boto3.resource("dynamodb")
+
+
+@functools.lru_cache(maxsize=None)
+def get_sqs_client() -> SQSClient:
+    return boto3.client("sqs")
+
+
+# OUTPUT_QUEUE_URL = os.environ["HASHES_QUEUE_URL"]
+DYNAMODB_TABLE = os.environ["DYNAMODB_TABLE"]
+
+# If you want to support additional content or signal types, they can be added
+# here.
```

Comment:
One thing I was expecting was a kind of "wildcard" catch, that would use a generic implementation of any signal that didn't have explicit support.

However, I assume that means we need equivalent fetch/sync support which we don't have yet, so guessing this a project for the future.